src/commands/build.rb

```diff
@@ -55,15 +55,28 @@ def run
 
       def dependency_hash
         {
-          repo: ["pxelinux/#{options.pxelinux}",
-                 "kickstart/#{options.kickstart}"],
+          repo: repo_dependencies,
           configure: ['domain.yaml'],
           optional: {
             configure: ["groups/#{group_name}.yaml"],
           },
         }
       end
 
+      def repo_dependencies
+        nodes.map do |node|
+          [:pxelinux, :kickstart].map do |template_type|
+            full_template_path = template_path(template_type, node: node)
+            repo_relative_path_to(full_template_path)
+          end
+        end.flatten.uniq
+      end
+
+      def repo_relative_path_to(path)
+        repo_path = Pathname.new(config.repo_path)
+        Pathname.new(path).relative_path_from(repo_path).to_s
+      end
```

Comment:
This method should be moved into `FilePath` class so it is available to other methods